### ISSUE TITLE
Progressbar doesn't print in single line

### ISSUE THREAD

**Issue by cloudnotify** [ctongfei/progressbar#82]
Kudos for the great utility @ctongfei , I had integrated progress bar with my application and the progress keep printing in new line(Tried in terminal and eclipse ).I'm using apache log4j2 for logging. Below is my code.

`
    	   

                    try (InputStream inputStream = new FileInputStream(source);
    	    		ProgressBar pb = new ProgressBarBuilder()
    	            .setInitialMax(inputStream.available())
    	            .setTaskName("Transfering")
    	            .setConsumer(new DelegatingProgressBarConsumer(logger
                    ::info)).setUnit("MB", 1048576)
    	            .build()) {
    		    

                    byte[] chunk = new byte[102400];
    		    int chunkLen = 0;
    		    

    		    while ((chunkLen = inputStream.read(chunk)) != -1) {
    		    	pb.stepBy(chunkLen);
    		    	session.getBasicRemote().sendBinary(ByteBuffer.wrap(chunk, 0, chunkLen));
    		    }
    		  
    	     
    	        } catch (IOException ex) {
    	        	
    	            logger.log(Level.ERROR,"Error ouccured while sending data from client ::",ex);
    	        }`

Also i have tried all the [visual styles](http://tongfei.me/progressbar/styles/). Any help is appreciated.

![test](https://user-images.githubusercontent.com/43980484/87751635-1a0e3680-c81c-11ea-9dc4-575c1835768d.png)

**Comment by ctongfei** [comment]
Maybe we should reopen this as this seems to be a valid concern?
It seems that it is because the header of the logger each line. Probably would be fixed if #71 is fixed, so that you can set a custom width?

**Comment by cloudnotify** [comment]
Thanks a lot @ctongfei, but i'm wondering how #71 is relevant to this. Here, the issue is progress-bar interaction with logger. :-)

**Comment by ctongfei** [comment]
@cloudnotify The progress bar has no knowledge of your logger header `[16 Jul 2020 22:59:57 INFO ] - `, so the computed width is wrong

**Comment by cloudnotify** [comment]
Thanks for the explanation @ctongfei

**Comment by tahifaMed** [comment]
Is there any Solution For this issue , I need some help, thank you

**Comment by ctongfei** [comment]
Fixed in `0.9.0`. You might want to do
```java
new DelegatingProgressBarConsumer(logger::info, <MAX WIDTH>)
```
where `<MAX WIDTH>` is the length of your terminal minus the length of your logger prefix (e.g. `[16 Jul 2020 22:59:57 INFO ] - `).
